src/include/storage/index/index_builder.h

```diff
@@ -67,7 +72,39 @@ class IndexBuilder {
     return *this;
   }
 
+ /**
+  * @param sql_table, transaction context
+  * @return the builder object
+  */
+  IndexBuilder &SetSqlTableAndTransactionContext(common::ManagedPointer<storage::SqlTable> sql_table,
+                                                 const common::ManagedPointer<transaction::TransactionContext> txn) {
+    TERRIER_ASSERT(sql_table == nullptr && txn == nullptr || sql_table != nullptr && txn != nullptr,
+                   "sql_table / txn is null and txn / sql_table is not.");
+    sql_table_ = sql_table;
+    txn_ = txn;
+    return *this;
+  }
+
  private:
+ /**
+  *
+  * @param newly created index
+  * @return index with all the keys inserted
+  */
+  void BulkInsert(Index *index) const {
+    uint32_t pr_size = index->GetProjectedRowInitializer().ProjectedRowSize();
+    byte *index_pr_buffer = common::AllocationUtil::AllocateAligned(pr_size);
+
+    ProjectedRow *index_pr = index->GetProjectedRowInitializer().InitializeRow(index_pr_buffer);
+
+    for (auto it = sql_table_->begin(); it != sql_table_->end(); ++it) {
+      const TupleSlot slot = *it;
+      index->Insert(txn_, *index_pr, slot);
+    }
+
+    delete[] index_pr_buffer;
+  }
```

Comment:
maybe in the future we can implement more efficient way to build indexed rather than one-by-one insertion